### PR TITLE
Add check-link to Makefile

### PR DESCRIPTION
**Pull request by bac**

### CODE REVIEW

**Comment by bac** [comment]
This change adds a target which uses linkchecker to look for broken links on the `devel` branch of docs on production.

```
make sysdeps
make check-link
```

**Comment by bac** [comment]
The docs currently have a number of issues:
```
linkchecker \
  --ignore-url=^https://jujucharms.com/docs/1.25 \
  --ignore-url=^https://jujucharms.com/docs/2.0 \
  --ignore-url=https://jujucharms.com/docs/stable \
  https://jujucharms.com/docs/devel/getting-started
LinkChecker 9.3              Copyright (C) 2000-2014 Bastian Kleineidam
LinkChecker comes with ABSOLUTELY NO WARRANTY!
This is free software, and you are welcome to redistribute it
under certain conditions. Look at the file `LICENSE' within this
distribution.
Get the newest version at http://wummel.github.io/linkchecker/
Write comments and bugs to https://github.com/wummel/linkchecker/issues
Support this project at http://wummel.github.io/linkchecker/donations.html

Start checking at 2016-08-30 12:22:16-004

URL        `./getting-started-faq'
Name       `FAQ'
Parent URL https://jujucharms.com/docs/devel/getting-started-general, line 718, col 1
Real URL   https://jujucharms.com/docs/devel/getting-started-faq/
Check time 3.669 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/getting-started-faq/'.
Result     Error: 404 Not Found

URL        `./charms-intro'
Name       `Charms/Services'
Parent URL https://jujucharms.com/docs/devel/getting-started-general, line 763, col 5
Real URL   https://jujucharms.com/docs/devel/charms-intro/
Check time 3.740 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/charms-intro/'.
Result     Error: 404 Not Found

URL        `stable/config-general'
Name       `General configuration options'
Parent URL https://jujucharms.com/docs/devel/clouds-LXD, line 712, col 8
Real URL   https://jujucharms.com/docs/devel/stable/config-general/
Check time 3.691 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/stable/config-general/'.
Result     Error: 404 Not Found

URL        `./models-destroying'
Name       `models documentation'
Parent URL https://jujucharms.com/docs/devel/charms-destroy, line 608, col 24
Real URL   https://jujucharms.com/docs/devel/models-destroying/
Check time 2.875 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/models-destroying/'.
Result     Error: 404 Not Found

URL        `config-general#alphabetical-list-of-general-configuration-values'
Name       `list of configuration parameters'
Parent URL https://jujucharms.com/docs/devel/juju-block, line 681, col 1
Real URL   https://jujucharms.com/docs/devel/config-general/
Check time 3.151 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/config-general/'.
Result     Error: 404 Not Found

URL        `./config-general#alphabetical-list-of-general-configuration-values'
Name       `"allow-lxc-loop-mounts"'
Parent URL https://jujucharms.com/docs/devel/charms-storage, line 817, col 3
Real URL   https://jujucharms.com/docs/devel/config-general/
Check time 3.756 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/config-general/'.
Result     Error: 404 Not Found

URL        `config-local'
Name       `local provider'
Parent URL https://jujucharms.com/docs/devel/howto-node, line 658, col 19
Real URL   https://jujucharms.com/docs/devel/config-local/
Check time 2.719 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/config-local/'.
Result     Error: 404 Not Found

URL        `config-aws'
Name       `AWS'
Parent URL https://jujucharms.com/docs/devel/howto-node, line 660, col 5
Real URL   https://jujucharms.com/docs/devel/config-aws/
Check time 2.568 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/config-aws/'.
Result     Error: 404 Not Found

URL        `config-hpcloud'
Name       `HP Cloud'
Parent URL https://jujucharms.com/docs/devel/howto-node, line 661, col 5
Real URL   https://jujucharms.com/docs/devel/config-hpcloud/
Check time 2.863 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/config-hpcloud/'.
Result     Error: 404 Not Found

URL        `./config-local'
Name       `local provider'
Parent URL https://jujucharms.com/docs/devel/howto-rails, line 657, col 19
Real URL   https://jujucharms.com/docs/devel/config-local/
Check time 3.007 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/config-local/'.
Result     Error: 404 Not Found

URL        `config-general'
Name       `General config options'
Parent URL https://jujucharms.com/docs/devel/config-openstack, line 705, col 8
Real URL   https://jujucharms.com/docs/devel/config-general/
Check time 3.756 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/config-general/'.
Result     Error: 404 Not Found

URL        `./config-hpcloud'
Name       `HP Cloud'
Parent URL https://jujucharms.com/docs/devel/howto-rails, line 660, col 5
Real URL   https://jujucharms.com/docs/devel/config-hpcloud/
Check time 3.373 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/config-hpcloud/'.
Result     Error: 404 Not Found

URL        `./config-aws'
Name       `AWS'
Parent URL https://jujucharms.com/docs/devel/howto-rails, line 659, col 5
Real URL   https://jujucharms.com/docs/devel/config-aws/
Check time 3.767 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/config-aws/'.
Result     Error: 404 Not Found

URL        `config-general'
Name       `General config options'
Parent URL https://jujucharms.com/docs/devel/config-openstack, line 705, col 8
Real URL   https://jujucharms.com/docs/devel/config-general/
Check time 3.756 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/config-general/'.
Result     Error: 404 Not Found

URL        `./config-vagrant'
Name       `Configuring for Vagrant'
Parent URL https://jujucharms.com/docs/devel/howto-vagrant-workflow, line 619, col 8
Real URL   https://jujucharms.com/docs/devel/config-vagrant/
Check time 3.551 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/config-vagrant/'.
Result     Error: 404 Not Found

URL        `./troubleshooting-juju'
Name       `Troubleshooting Juju itself'
Parent URL https://jujucharms.com/docs/devel/troubleshooting, line 610, col 5
Real URL   https://jujucharms.com/docs/devel/troubleshooting-juju/
Check time 3.728 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/troubleshooting-juju/'.
Result     Error: 404 Not Found

URL        `./charms-layers-interfaces.md'
Name       `interface-layers'
Parent URL https://jujucharms.com/docs/devel/developer-getting-started, line 681, col 1
Real URL   https://jujucharms.com/docs/devel/charms-layers-interfaces/
Check time 3.827 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/charms-layers-interfaces'.
           Redirected to
           `https://jujucharms.com/docs/devel/charms-layers-interfaces/'.
Result     Error: 404 Not Found

URL        `./charms-layers-interfaces.md'
Name       `interface-layers'
Parent URL https://jujucharms.com/docs/devel/developer-getting-started#install-libraries-and-tools, line 681, col 1
Real URL   https://jujucharms.com/docs/devel/charms-layers-interfaces/
Check time 3.866 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/charms-layers-interfaces'.
           Redirected to
           `https://jujucharms.com/docs/devel/charms-layers-interfaces/'.
Result     Error: 404 Not Found

URL        `./charms-layers-interfaces.md'
Name       `interface-layers'
Parent URL https://jujucharms.com/docs/devel/developer-getting-started#writing-your-charm, line 681, col 1
Real URL   https://jujucharms.com/docs/devel/charms-layers-interfaces/
Check time 3.465 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/charms-layers-interfaces'.
           Redirected to
           `https://jujucharms.com/docs/devel/charms-layers-interfaces/'.
Result     Error: 404 Not Found

URL        `./charms-layers-interfaces.md'
Name       `interface-layers'
Parent URL https://jujucharms.com/docs/devel/developer-getting-started#designing-your-charm, line 681, col 1
Real URL   https://jujucharms.com/docs/devel/charms-layers-interfaces/
Check time 3.520 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/charms-layers-interfaces'.
           Redirected to
           `https://jujucharms.com/docs/devel/charms-layers-interfaces/'.
Result     Error: 404 Not Found

URL        `./charms-layers-interfaces.md'
Name       `interface-layers'
Parent URL https://jujucharms.com/docs/devel/developer-getting-started#testing-your-charm, line 681, col 1
Real URL   https://jujucharms.com/docs/devel/charms-layers-interfaces/
Check time 3.814 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/charms-layers-interfaces'.
           Redirected to
           `https://jujucharms.com/docs/devel/charms-layers-interfaces/'.
Result     Error: 404 Not Found

URL        `./charms-layers-interfaces.md'
Name       `interface-layers'
Parent URL https://jujucharms.com/docs/devel/developer-layers-interfaces, line 686, col 1
Real URL   https://jujucharms.com/docs/devel/charms-layers-interfaces/
Check time 3.620 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/charms-layers-interfaces'.
           Redirected to
           `https://jujucharms.com/docs/devel/charms-layers-interfaces/'.
Result     Error: 404 Not Found

URL        `./charms-layers-interfaces.md'
Name       `interface-layers'
Parent URL https://jujucharms.com/docs/devel/developer-layer-example, line 626, col 1
Real URL   https://jujucharms.com/docs/devel/charms-layers-interfaces/
Check time 3.967 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/charms-layers-interfaces'.
           Redirected to
           `https://jujucharms.com/docs/devel/charms-layers-interfaces/'.
Result     Error: 404 Not Found

URL        `./charms-layers-interfaces.md'
Name       `interface-layers'
Parent URL https://jujucharms.com/docs/devel/developer-layers-interfaces#design-considerations, line 686, col 1
Real URL   https://jujucharms.com/docs/devel/charms-layers-interfaces/
Check time 4.064 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/charms-layers-interfaces'.
           Redirected to
           `https://jujucharms.com/docs/devel/charms-layers-interfaces/'.
Result     Error: 404 Not Found

URL        `./charms-layers-interfaces.md'
Name       `interface-layers'
Parent URL https://jujucharms.com/docs/devel/developer-layers-interfaces#writing-an-interface-layer, line 686, col 1
Real URL   https://jujucharms.com/docs/devel/charms-layers-interfaces/
Check time 3.364 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/charms-layers-interfaces'.
           Redirected to
           `https://jujucharms.com/docs/devel/charms-layers-interfaces/'.
Result     Error: 404 Not Found

URL        `./charms-layers-interfaces.md'
Name       `interface-layers'
Parent URL https://jujucharms.com/docs/devel/developer-layers-interfaces#communication-scopes, line 686, col 1
Real URL   https://jujucharms.com/docs/devel/charms-layers-interfaces/
Check time 4.074 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/charms-layers-interfaces'.
           Redirected to
           `https://jujucharms.com/docs/devel/charms-layers-interfaces/'.
Result     Error: 404 Not Found

URL        `authors-charm-subordinates'
Name       `Subordinate'
Parent URL https://jujucharms.com/docs/devel/authors-relations, line 689, col 1
Real URL   https://jujucharms.com/docs/devel/authors-charm-subordinates/
Check time 3.720 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/authors-charm-subordinates/'.
Result     Error: 404 Not Found

URL        `devel/authors-charm-store#entities-explained'
Name       `documentation'
Parent URL https://jujucharms.com/docs/devel/temp-release-notes, line 1392, col 1
Real URL   https://jujucharms.com/docs/devel/devel/authors-charm-store/
Check time 3.140 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/devel/authors-charm-store/'.
Result     Error: 404 Not Found

URL        `./juju-config.md'
Name       `see config docs'
Parent URL https://jujucharms.com/docs/devel/command-changes, line 774, col 30
Real URL   https://jujucharms.com/docs/devel/juju-config/
Check time 2.926 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/juju-config'.
           Redirected to
           `https://jujucharms.com/docs/devel/juju-config/'.
Result     Error: 404 Not Found

URL        `./juju-config.md'
Name       `see config docs'
Parent URL https://jujucharms.com/docs/devel/command-changes, line 798, col 30
Real URL   https://jujucharms.com/docs/devel/juju-config/
Check time 2.833 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/juju-config'.
           Redirected to
           `https://jujucharms.com/docs/devel/juju-config/'.
Result     Error: 404 Not Found

URL        `./developers-charm-store'
Name       `see docs on publishing charms'
Parent URL https://jujucharms.com/docs/devel/command-changes, line 814, col 30
Real URL   https://jujucharms.com/docs/devel/developers-charm-store/
Check time 2.784 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/developers-charm-store/'.
Result     Error: 404 Not Found

URL        `./juju-backup-restore'
Name       `Backup and restore'
Parent URL https://jujucharms.com/docs/devel/models-upgrade, line 637, col 3
Real URL   https://jujucharms.com/docs/devel/juju-backup-restore/
Check time 3.035 seconds
Size       23.85KB
Info       Redirected to
           `https://jujucharms.com/docs/devel/juju-backup-restore/'.
Result     Error: 404 Not Found

URL        `./authors-\ncharm-components'
Name       `charm components'
Parent URL https://jujucharms.com/docs/devel/authors-charm-writing, line 750, col 55
Real URL   https://jujucharms.com/docs/devel/authors-%0Acharm-components
Check time 2.917 seconds
Size       314B
Result     Error: 404 Not Found

URL        `./authors-\ncharm-interfaces'
Name       `more information on decoding interfaces here'
Parent URL https://jujucharms.com/docs/devel/authors-charm-writing, line 897, col 22
Real URL   https://jujucharms.com/docs/devel/authors-%0Acharm-interfaces
Check time 2.991 seconds
Size       314B
Result     Error: 404 Not Found

URL        `./authors-charm-\nupgrades'
Name       `upgrade'
Parent URL https://jujucharms.com/docs/devel/authors-charm-hooks#%5Bname%5D-relation-broken, line 666, col 58
Real URL   https://jujucharms.com/docs/devel/authors-charm-%0Aupgrades
Check time 3.600 seconds
Size       312B
Result     Error: 404 Not Found

URL        `./authors-\ncharm-components'
Name       `charm components'
Parent URL https://jujucharms.com/docs/devel/authors-charm-writing, line 750, col 55
Real URL   https://jujucharms.com/docs/devel/authors-%0Acharm-components
Check time 2.917 seconds
Size       314B
Result     Error: 404 Not Found

URL        `./authors-\ncharm-interfaces'
Name       `more information on decoding interfaces here'
Parent URL https://jujucharms.com/docs/devel/authors-charm-writing, line 897, col 22
Real URL   https://jujucharms.com/docs/devel/authors-%0Acharm-interfaces
Check time 2.991 seconds
Size       314B
Result     Error: 404 Not Found

URL        `./authors-\ncharm-best-practice'
Name       `best practices'
Parent URL https://jujucharms.com/docs/devel/authors-charm-hooks#%5Bname%5D-relation-broken, line 786, col 56
Real URL   https://jujucharms.com/docs/devel/authors-%0Acharm-best-practice
Check time 2.499 seconds
Size       317B
Result     Error: 404 Not Found

URL        `./authors-charm-\nmetadata'
Name       `metadata'
Parent URL https://jujucharms.com/docs/devel/authors-charm-hooks#%5Bname%5D-relation-broken, line 697, col 63
Real URL   https://jujucharms.com/docs/devel/authors-charm-%0Ametadata
Check time 3.052 seconds
Size       312B
Result     Error: 404 Not Found

URL        `./authors-charm-\nupgrades'
Name       `upgrade'
Parent URL https://jujucharms.com/docs/devel/authors-charm-hooks#%5Bname%5D-relation-changed, line 666, col 58
Real URL   https://jujucharms.com/docs/devel/authors-charm-%0Aupgrades
Check time 3.647 seconds
Size       312B
Result     Error: 404 Not Found

URL        `./authors-\ncharm-best-practice'
Name       `best practices'
Parent URL https://jujucharms.com/docs/devel/authors-charm-hooks#%5Bname%5D-relation-changed, line 786, col 56
Real URL   https://jujucharms.com/docs/devel/authors-%0Acharm-best-practice
Check time 3.287 seconds
Size       317B
Result     Error: 404 Not Found

URL        `./authors-charm-\nmetadata'
Name       `metadata'
Parent URL https://jujucharms.com/docs/devel/authors-charm-hooks#%5Bname%5D-relation-changed, line 697, col 63
Real URL   https://jujucharms.com/docs/devel/authors-charm-%0Ametadata
Check time 3.901 seconds
Size       312B
Result     Error: 404 Not Found

URL        `./authors-charm-\nmetadata'
Name       `metadata'
Parent URL https://jujucharms.com/docs/devel/authors-charm-hooks, line 697, col 63
Real URL   https://jujucharms.com/docs/devel/authors-charm-%0Ametadata
Check time 3.278 seconds
Size       312B
Result     Error: 404 Not Found

URL        `./authors-charm-\nupgrades'
Name       `upgrade'
Parent URL https://jujucharms.com/docs/devel/authors-charm-hooks, line 666, col 58
Real URL   https://jujucharms.com/docs/devel/authors-charm-%0Aupgrades
Check time 4.094 seconds
Size       312B
Result     Error: 404 Not Found

URL        `./authors-\ncharm-best-practice'
Name       `best practices'
Parent URL https://jujucharms.com/docs/devel/authors-charm-hooks, line 786, col 56
Real URL   https://jujucharms.com/docs/devel/authors-%0Acharm-best-practice
Check time 4.385 seconds
Size       317B
Result     Error: 404 Not Found

Statistics:
Downloaded: 8.52MB.
Content types: 0 image, 25957 text, 0 video, 0 audio, 0 application, 2 mail and 14112 other.
URL lengths: min=14, max=256, avg=52.

That's it. 40071 links in 166 URLs checked. 0 warnings found. 45 errors found.
Stopped checking at 2016-08-30 12:23:33-004 (1 minute, 17 seconds)
Makefile:29: recipe for target 'check-links' failed
```

File: Makefile [hunk 1]
```diff
@@ -12,8 +12,9 @@ serve:
 todo:
 	tools/mdbuild.py --todo
 sysdeps:
-	sudo apt-get install python-html2text python3-markdown python3-pip git spell ispell ibritish python3-setuptools
+	sudo apt-get install python-html2text python3-markdown python-pip python3-pip git spell ispell ibritish python3-setuptools
 	sudo pip3 install mdx-anchors-away mdx-callouts mdx-foldouts
+	sudo pip install linkchecker
```
Comment (bac) [inline Makefile]: linkchecker is not available for python3

**Comment by evilnick** [comment]
Thanks. We are actually putting a link-checker into the parser code which will emit warning etc (and maybe one day we will CI it), but this is useful for now. Thanks!

**Pull request merged**